src/sentry/services/hybrid_cloud/audit/impl.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from typing import Any, Mapping
+
+from sentry.models import AuditLogEntry, AuthIdentity, AuthProvider, Organization, Team, User
+from sentry.services.hybrid_cloud.audit import AuditLogMetadata, AuditLogService
+from sentry.services.hybrid_cloud.auth import ApiAuthIdentity, ApiAuthProvider
+from sentry.services.hybrid_cloud.organization import ApiOrganizationMember
+from sentry.services.hybrid_cloud.user import APIUser, user_service
+
+
+class DatabaseBackedAuditLogService(AuditLogService):
+    @staticmethod
+    def _resolve_user(user: APIUser | int | None) -> User | None:
+        if user is None:
+            return None
+        if isinstance(user, int):
+            user_id = user
+        else:
+            user_id = user.id
+        return User.objects.get(id=user_id)
+
+    def write_audit_log(self, *, metadata: AuditLogMetadata, data: Mapping[str, Any]) -> None:
+        organization = Organization.objects.get(id=metadata.organization.id)
+        AuditLogEntry.objects.create(
+            organization=organization,
+            event=metadata.event,
+            actor_label=metadata.actor_label,
+            actor=self._resolve_user(metadata.actor),
```

Comment:
Does this need the full actor object?  Would it be sufficient to pass an id alone?  Also @vbro for 👀

Reply:
That sounds right. I'll experiment.